web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/SelectSecurityGroups.tsx

```diff
@@ -307,6 +312,6 @@ function ruleAllowsPort(rule: SecurityGroupRule, port: number): boolean {
 }
 
 function getPort(db: AwsRdsDatabase): number {
-  const [, port = '-1'] = db.uri.split(':');
+  const [, port = '-1'] = db?.uri?.split(':') ?? [];
```

Comment:
i think we can expect database to be defined at this step right? (i'd remove the checks)

it's actually not good to put `?` everywhere, we want to be intentional, like if we are sure if it's an optional field then yes use `?`, if we are not, we have to look into why we aren't sure (i'm pretty sure it's my fault for typing things badly)

i think going forward we need to write painful UI test for each individual flow and scenarios to at least test that things are being rendered correctly

Reply:
I check for undefined before passing the db into this func now